American Whitewater/ViewControllers/Reaches/RunMapViewController.swift

```diff
@@ -0,0 +1,267 @@
+                guard let rapids = annotation.reach?.rapids else { return }
+
+                let rapidMatch = rapids.filter {
+                    if let rapid = $0 as? Rapid {
+                        return rapid.name == annotation.title
+                    } else {
+                        return false
+                    }
+                }
+
+                if let rapid = rapidMatch.first {
+                    performSegue(withIdentifier: "mapRapidDetails", sender: rapid)
+                }
+            }
+        }
+    }
+    
+    
+    func mapView(_ mapView: MKMapView, didUpdate userLocation: MKUserLocation) {
+        let userLat = userLocation.coordinate.latitude
+        let userLon = userLocation.coordinate.longitude
+        
+        //print("Location difference is: Map: \(userLat) - DefLat: \(DefaultsManager.latitude) = \(abs(userLat - DefaultsManager.latitude)) x \(userLon) - \(DefaultsManager.longitude) = \(abs(userLon - DefaultsManager.longitude))")
+        
+        // check if we need to update locations
+        if abs(userLat - DefaultsManager.latitude) > 0.01 ||
+            abs(userLon - DefaultsManager.longitude) > 0.01 {
+            print("Updating distances of reaches")
+            
+            DefaultsManager.latitude = userLat
+            DefaultsManager.longitude = userLon
+        }
+
+        self.mapView.setVisibleMapRectToFitAllAnnotations(animated: true, shouldIncludeUserAccuracyRange: true, shouldIncludeOverlays: true)
+    }
+    
+    
+    func showUserLocation() {
+        mapView.showsUserLocation = true
+        
+        self.mapView.setVisibleMapRectToFitAllAnnotations(animated: true, shouldIncludeUserAccuracyRange: true, shouldIncludeOverlays: true)
+    }
+    
+    func checkLocationAuthorizationStatus() {
+        if CLLocationManager.authorizationStatus() == .authorizedWhenInUse {
+            // it's approved - button will now work
+            return
+        } else if CLLocationManager.authorizationStatus() == .denied {
+            showLocationDeniedMessage()
+        } else {
+            locationManager.requestWhenInUseAuthorization()
+        }
+    }
+
+    func locationManager(_ manager: CLLocationManager, didChangeAuthorization status: CLAuthorizationStatus) {
+        switch status {
+            case .authorizedWhenInUse:
+                //showUserLocation()
+                break
+            case .authorizedAlways:
+                // showUserLocation()
+                break
+            case .denied:
+                showLocationDeniedMessage()
+            default:
+                break
+        }
+    }
+    
+    func showLocationDeniedMessage() {
+        DuffekDialog.shared.showStandardDialog(title: "Permission Denied", message: "You chose to deny this app location permissions and we are unable to use your current location for displaying the map. Please update your settings and try again.", buttonTitle: "Change Settings", buttonFunction: {
```

Comment:
Shouldn't these strings be stored centrally somewhere else? I've seen them in multiple places.